docs/cancellation.rst

```diff
@@ -209,24 +214,23 @@ start to end in the same task, making it possible to have task groups or cancel
 safely straddle the ``yield``.
 
 When you're implementing the async context manager protocol manually and your async
-context manager needs to use other context managers, you may find it necessary to call
-their ``__aenter__()`` and ``__aexit__()`` directly. In such cases, it is absolutely
-vital to ensure that their ``__aexit__()`` methods are called in the exact reverse order
-of the ``__aenter__()`` calls. To this end, you may find the
-:class:`~contextlib.AsyncExitStack` class very useful::
+context manager needs to use other context managers, you may find it convenient to use
+:class:`AsyncContextManagerMixin` in order to avoid cumbersome code that calls
+``__aenter__()`` and ``__aexit__()`` directly::
+
+    from __future__ import annotations
 
-    from contextlib import AsyncExitStack
+    from collections.abc import AsyncGenerator
+    from typing import Self
 
-    from anyio import create_task_group
+    from anyio import AsyncContextManagerMixin, create_task_group
 
 
-    class MyAsyncContextManager:
-        async def __aenter__(self):
-            self._exitstack = AsyncExitStack()
-            await self._exitstack.__aenter__()
```

Comment:
oh this was slightly wrong it should have used pop_all()

I think it's worth including an ExitStack instruction on how to do this